File: lib/src/models/measure.dart
import 'package:app/src/utils/date_converter.dart';

class Measure {
  Measure({
    required this.current,
    required this.power,
    required this.timestamp,
    required this.voltage,
  });

  double current;
  double power;
  String timestamp;
  double voltage;

  factory Measure.fromJson(Map<String, dynamic> json) => Measure(
        current: json["current"].toDouble(),
        power: json["power"].toDouble(),
        timestamp: json["timestamp"],
        voltage: json["voltage"].toDouble(),
      );

  DateTime get datetime {
    return DateConverter.fromISOString(this.timestamp);
  }

  double get doubleTimestamp {
    return DateConverter.toTimestamp(this.datetime);
  }

  Measure clone() {
    return new Measure(
        current: current, power: power, timestamp: timestamp, voltage: voltage);
  }
}

File: src/models/measure.ts
export class Measure {
    protected _current: number;
    protected _voltage: number;
    protected _power: number;
    protected _timestamp: Date;

    constructor(current: number, voltage: number, power: number, timestamp: Date) {
        this._current = current;
        this._voltage = voltage;
        this._power = power;
        this._timestamp = timestamp;
    }


    public get current(): number {
        return this._current;
    }

    public get voltage(): number {
        return this._voltage;
    }

    public get power(): number {
        return this._power;
    }

    public get timestamp(): Date {
        return this._timestamp;
    }


    public static fromObject(obj: any): Measure {
        return new Measure(
            obj.current,
            obj.voltage,
            obj.power,
            new Date(Date.parse(obj.timestamp as string))
        );
    }
}